Src/Newtonsoft.Json/JsonTextWriter.cs

```diff
@@ -784,10 +784,20 @@ public override void WriteValue(Uri? value)
         public override void WriteComment(string? text)
         {
             InternalWriteComment();
-
-            _writer.Write("/*");
-            _writer.Write(text);
-            _writer.Write("*/");
+            
+            // if text contains "*/" then it must have been a line comment
+            if (text != null && text.IndexOf("*/", StringComparison.Ordinal) > -1)
+            {
+                _writer.Write("//");
+                _writer.Write(text);
```

Comment:
What happens if text has `*/` and a new line?

Reply:
This shouldn't be possible via parsing but if you instantiate a comment type directly you can do it. I'll add some explicit test cases for this.